sentry/src/test/java/io/sentry/HubTest.kt

```diff
@@ -1312,6 +1281,61 @@ class HubTest {
     }
     //endregion
 
+    //region captureProfile tests
+    @Test
+    fun `when captureProfile is called on disabled client, do nothing`() {
+        val options = SentryOptions()
+        options.cacheDirPath = file.absolutePath
+        options.dsn = "https://key@sentry.io/proj"
+        options.setSerializer(mock())
+        val sut = Hub(options)
+        val mockClient = mock<ISentryClient>()
+        sut.bindClient(mockClient)
+        sut.close()
+
+        val sentryTracer = SentryTracer(TransactionContext("name", "op"), sut)
+        sentryTracer.finish()
+        sut.captureProfile(ProfilingTraceData(profilingTraceFile, sentryTracer))
+        verify(mockClient, never()).captureProfile(any())
+    }
+
+    @Test
+    fun `when startTransaction and profiling is enabled, transaction is profiled only if sampled`() {
+        val mockTransactionProfiler = mock<ITransactionProfiler>()
+        val mockClient = mock<ISentryClient>()
+        val hub = generateHub {
+            it.setTransactionProfiler(mockTransactionProfiler)
+        }
+        hub.bindClient(mockClient)
+        // Transaction is not sampled, so it should not be profiled
+        val contexts = TransactionContext("name", "op", TracesSamplingDecision(false, null, true, null))
+        val transaction = hub.startTransaction(contexts)
+        transaction.finish()
+        verify(mockClient, never()).captureProfile(any())
+
+        // Transaction is sampled, so it should be profiled
+        val sampledContexts = TransactionContext("name", "op", TracesSamplingDecision(true, null, true, null))
+        val sampledTransaction = hub.startTransaction(sampledContexts)
+        sampledTransaction.finish()
+        verify(mockClient, never()).captureProfile(any())
```

Comment:
`m`: Shouldn't this be
```suggestion
        verify(mockClient).captureProfile(any())
```